OPHD/States/MapViewStateTurn.cpp

```diff
@@ -277,8 +277,9 @@ void MapViewState::updateResources()
 			std::clamp(stored.resources[3], 0, 25)
 		};
 
+		stored -= moved;
 		addRefinedResources(moved);
-		stored = stored + moved;
+		stored += moved;
```

Comment:
It looks odd but it works like this -- the amount moved (`moved` is the maximum amount that could be moved) is subtracted from the smelter storage. `addRefinedResources()` is called which modifies `moved` in place (though I suppose this could return the remainder instead of modifying its parameter). After `addRefinedResources()` is called, `moved` may be empty or their may be some overflow. This is then added back to `stored`. `addRefinedResources()` adds as much as possible from moved into storage though not all of it may fit into storage. I could be persuaded to change the function name if it needs it (like, `addRefinedResourcesToStorage()` or really whatever we think is appropriate).

If `addRefinedResources()` is modified to return the overflow instead of modifying its parameter, the code could read like this:

```cpp
stored -= moved;
stored += addRefinedResources(moved);
```

Might be a little easier to understand

Reply:
I like your suggestions here. That would make the code more clear.